### PR TITLE
[snapshot] Integration tests: archive Elastic stack logs

### PR DESCRIPTION
**Pull request by mtojek**
This PR modifies the "Setup" integration test to archive all container logs as Jenkins artifacts.

Link to artifacts: https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fpackage-storage/detail/PR-759/3/artifacts

### CODE REVIEW

**Review by kuisathaverat — APPROVED** [review]
we will release a pipeline step to make more or less the same https://github.com/elastic/apm-pipeline-library/pull/887

**Comment by mtojek** [comment]
@kuisathaverat not sure if I follow your PR. The description doesn't look relevant. Would  you mind elaborating more about it?

**Comment by kuisathaverat** [comment]
> Ok, I think I got the idea. You're starting additional container with filebeat to dump all logs. Out of curiosity: if the enclosed code (in this situation: integrationTests) starts and stops containers, will filebeat notice and dump logs also from that instances?

yes, every log event is captured in that file. The default configuration uses a single file, but it is possible to pass a custom configuration file.

> 
> The problem with your solution is that it can't be run locally e.g. while root cause analysis or trying to reproduce an issue. The developer would have to run filebeat on their own.

yes, the step is designed for CI only, however, you can run filebeat with a similar configuration in your docker-compose and rid of all the go logic to manage those logs.

File: testing/main_integration_test.go [hunk 1]
```diff
@@ -12,42 +12,35 @@ import (
...
 	"net/http"
 	"os"
+	"path/filepath"
 	"strings"
 	"testing"
 	"time"
 
 	"github.com/magefile/mage/sh"
-	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
```
Comment (ycombinator) [inline testing/main_integration_test.go:21]: ❤️  👍

**Review by ycombinator — APPROVED** [review]
LGTM.

**Pull request merged**